taichi/ir/frontend_ir.cpp

```diff
@@ -919,10 +919,10 @@ void AtomicOpExpression::flatten(FlattenContext *ctx) {
     op_type = AtomicOpType::add;
   }
   // expand rhs
-  flatten_rvalue(val, ctx);
-  auto src_val = val->stmt;
-  flatten_lvalue(dest, ctx);
-  stmt = ctx->push_back<AtomicOpStmt>(op_type, dest->stmt, src_val);
+  auto val_stmt = flatten_rvalue(val, ctx);
+  auto src_val = val_stmt;
+  auto dest_stmt = flatten_lvalue(dest, ctx);
+  stmt = ctx->push_back<AtomicOpStmt>(op_type, dest_stmt, src_val);
```

Comment:
```suggestion
  auto val_stmt = flatten_rvalue(val, ctx);
  auto dest_stmt = flatten_lvalue(dest, ctx);
  stmt = ctx->push_back<AtomicOpStmt>(op_type, dest_stmt, val_stmt);
```
After the refactoring I think we can eliminate
```cpp
auto b = a;
c = b;
```
to make the code cleaner. It seems that there are a few places with this situation.